### PR TITLE
Move thread resolve priv check to backend

### PR DESCRIPTION
**Pull request by nanaya**
And allow BNG to resolve as well.

Related to #1609? ┐(' ヮ' )┌

### CODE REVIEW

File: app/Models/Model.php [hunk 1]
```diff
@@ -59,4 +59,12 @@ public function saveOrExplode($options = [])
 
         return $result;
     }
+
+    public function save(array $options = [])
+    {
+        $result = parent::save($options);
+        priv_clear_cache();
```
Comment (nekodex) [inline app/Models/Model.php]: I feel like this shouldn't be on the base model? Should just call it as needed? A lot of unnecessary clearing is gonna happen... (also why is this even needed?)

Comment (nanaya) [reply app/Models/Model.php]: Changes to model may change authorization.

Comment (peppy) [reply app/Models/Model.php]: which models? shouldn't it only be done where it is necessary?

Comment (nanaya) [reply app/Models/Model.php]: there's currently only manual way to figure out whether it is necessary.

1. priv_check on a model
2. update the model
3. priv_check again for view

If `2` changes result of priv check, the cache must be cleared at one point before `3` is done. Placing it in controller can also work but it means either clearing the cache everywhere before rendering the view (incl. transformer) or checking whether priv check is used by the view, already cached, and affected by the just updated model.

**Comment by nekodex** [comment]
Admins still seem to bypass the `$discussion->canBeResolved()` check, allowing them to 'resolve' praises and stuff still?

**Comment by nanaya** [comment]
no

**Comment by nekodex** [comment]
oh wait branch wasn't up to date, nvm derp

**Pull request merged**